runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilityPlugin.java

```diff
@@ -366,11 +382,25 @@ private void onTileObject(Tile tile, TileObject oldObject, TileObject newObject)
 		}
 
 		if (Obstacles.COURSE_OBSTACLE_IDS.contains(newObject.getId()) ||
-			Obstacles.SHORTCUT_OBSTACLE_IDS.contains(newObject.getId()) ||
 			(Obstacles.TRAP_OBSTACLE_IDS.contains(newObject.getId())
 				&& Obstacles.TRAP_OBSTACLE_REGIONS.contains(newObject.getWorldLocation().getRegionID())))
 		{
-			obstacles.put(newObject, tile);
+			obstacles.put(newObject, new Obstacle(tile, 0));
+		}
+
+		if (Obstacles.SHORTCUT_OBSTACLE_IDS.contains(newObject.getId()))
+		{
+			Stream.of(AgilityShortcut.values())
+				.forEach(obstacle ->
+				{
+					if (obstacle.getObstacleIds().contains(newObject.getId())
+							&& (obstacle.getShortcutBounds() == null
+							|| (obstacle.getShortcutBounds() != null && obstacle.getShortcutBounds().contains(newObject.getWorldLocation().getX(), newObject.getWorldLocation().getY())))
```

Comment:
The comparison `obstacle.getShortcutBounds() != null` is always true here

Reply:
Not all obstacles have shortcut bounds. It's only used when there's duplicate IDs in the same area.

Reply:
Adam means it is always true because otherwise the `obstacle.getShortcutBounds() == null` on the line above this one would be `true`, and short-circuit, causing this line not to run at all.